docs/proposals/20240807-in-place-updates.md

```diff
@@ -353,43 +359,36 @@ box Workload Cluster
-loop Until finished
+loop For all External Updaters
     mach->>hook: finished?
-    hook->>mach: try in X secs
-end
-mach->>apiserver: Remove Updater 1 from plan
-mach->>hook2: Start update
-hook2->>infra: Update components
-loop Until finished
-    mach->>hook2: finished?
     hook2->>mach: try in X secs
 end
-mach->>apiserver: Remove Updater 2 from plan
-mach->>apiserver: Mark Machine as up to date
-
+mach->>apiserver: Mark Machine as Done
+capi->>apiserver: Set UpToDate condition
 ```
 
-Once a the Machine update plan is set and the Machine's spec has been updated with the desired changes, the Machine controller takes over. This controller is responsible for executing the right updaters in the right order, tracking the progress of those updaters and exposing this progress in the Machine status.
+Once a Machine is marked as pending and `UpToDate` condition is set and the Machine's spec has been updated with the desired changes, the Machine controller takes over. This controller is responsible for calling the updaters and tracking the progress of those updaters and exposing this progress in the Machine conditions.
 
-The Machine controller will follow the order in which the names of the updaters have been defined in the Machine object, executing them one by one, sequentially.
+The Machine controller will not follow any order when calling the updaters. This might change in future iterations.
 
-The controller will trigger updaters by hitting another RuntimeHook endpoint (eg. `/UpdateMachine`). This endpoint will take the Machine name as input. The updater could respond saying "update completed", "update failed" or "update in progress" with an optional "retry after X seconds". The CAPI controller will continuously poll the status of the update by hitting the same endpoint until it reaches a terminal state.
+The controller will trigger updaters by hitting another RuntimeHook endpoint (eg. `/UpdateMachine`). The updater could respond saying "update completed", "update failed" or "update in progress" with an optional "retry after X seconds". The CAPI controller will continuously poll the status of the update by hitting the same endpoint until it reaches a terminal state.
 
 CAPI expects the `/UpdateMachine` endpoint of an updater to be idempotent: for the same Machine with the same spec, the endpoint can be called any number of times (before and after it completes), and the end result should be the same. CAPI guarantees that once an `/UpdateMachine` endpoint has been called once, it won't change the Machine spec until the update reaches a terminal state.
 
-Once the update completes, the Machine controller will remove the name of the updater that has finished from the list of updaters and will start the next one. If the update fails, this will be reflected in the Machine status.
+Once the update completes, the Machine controller will mark machine as done. If the update fails, this will be reflected in the Machine status.
```

Comment:
```suggestion
Once the update completes, the Machine controller will mark machine's hooks as done. If the update fails, this will be reflected in the Machine status.
```